tasks/assembl.py

```
@@ -70,9 +91,204 @@ def create_var_dir(c):
+        template = os.path.join(default_dir, template)
+    if not os.path.exists(template):
+        raise RuntimeError("Missing template")
+    c.config['here'] = c.config.get('here', os.getcwd())
+    if template.endswith('.tmpl'):
+        with open(template) as tmpl:
+            result = tmpl.read() % c.config
+    elif template.endswith('.jinja2'):
+        from jinja2 import Environment
+        env = Environment()
+        with open(template) as tmpl:
+            tmpl = env.from_string(tmpl.read())
+        # Boolean overloading
+        # Jinja should interpret 'false' as False but no:
+        # https://github.com/ansible/ansible/issues/14983
+        for (k, v) in c.config.items():
+            if str(v).lower() == 'false':
+                c.config[k] = False
+            if '%(' in str(v):
+                try:
+                    c.config[k] = v % c.config
+                except KeyError:
+                    pass
+        result = tmpl.render(c.config)
+    else:
+        raise RuntimeError("Unknown template type")
+    if hasattr(output, 'write'):
+        output.write(result)
+    else:
+        with open(output, 'w') as out:
+            out.write(result)
+
+
+def is_supervisord_running(c):
+    with venv(c):
+        result = c.run('supervisorctl pid')
+    if 'no such file' in result.stdout:
+        return False
+    try:
+        pid = int(result.stdout)
+        if pid:
+            return True
+    except RuntimeError:
+        return False
+
+
+@task()
+def supervisor_shutdown(c):
+    """
+    Shutdown Assembl's supervisor
+    """
+    with venv(c):
+        if not is_supervisord_running(c):
+            return
+        c.run("supervisorctl shutdown")
+        for i in range(10):
+            sleep(6)
+            result = c.run("supervisorctl status", warn=True)
+            if not result.failed:
+                break
+    # Another supervisor, upstart, etc may be watching it, give it more time
+    sleep(6)
```

Here, you unsure if supervisor is down or not. Maybe raise an exception if after all retries it is still up ? Or kill the process ?